File: 0506-relative-ranks/0506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        n = len(score)
        pq, ans, awards = [], [""] * n, []
        
        for i in range(n):
            heappush(pq, (-score[i], i))
            
            if i == 0:
                awards.append("Gold Medal")
            elif i == 1:
                awards.append("Silver Medal")
            elif i == 2:
                awards.append("Bronze Medal")
            else:
                awards.append(str(i + 1))
            
        curr = 0
        while pq:
            rank, idx = heappop(pq)
            ans[idx] = awards[curr]
            curr += 1
        
        return ans

File: 0506-relative-ranks/0506-relative-ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        int n = score.size();
        priority_queue<pair<int, int>> pq;
        vector<string> ans(n), awards;
        
        for (int i = 0; i < n; ++i){
            pq.push({score[i], i});
            if (i == 0)
                awards.push_back("Gold Medal");
            else if (i == 1)
                awards.push_back("Silver Medal");
            else if (i == 2)
                awards.push_back("Bronze Medal");
            else awards.push_back(to_string(i + 1));
        }
        
        int curr = 0;
        while (!pq.empty()){
            auto [rank, idx] = pq.top(); pq.pop();
            ans[idx] = awards[curr++];
        }
        
        return ans;
    }
};